Add other plot save formats

timtmok opened the pull request:
### Intent
Address #2657 

### Approach
Adds a format option when selecting where to save the plot. This adds a new parameter to the render request and updates the Python extension to render the other formats. Ark will have its own PR to add in that support. Until Ark supports the other formats, plot saving will always save as a PNG regardless of the chosen format.

![image](https://github.com/posit-dev/positron/assets/9591545/98ba6f4c-80be-4d26-9612-ec5ebad58572)

### QA Notes
Sample plot for Python
```python
import numpy as np
import matplotlib.pyplot as plt

with plt.xkcd():
    # plt in widget mode
    # %matplotlib ipympl
    # v = np.array([1, 3, 5, 365])
    v = np.arange(365, 0, -1)
    # w = np.array([4, 6, 2, 10, 8, 1])

    fig, ax = plt.subplots()
    fig.set_facecolor('whitesmoke')
    fig.set_edgecolor('black')
    ax.set_xlabel('Day of the year')
    ax.set_ylabel('Days until New Year\'s Eve')

    plt.xlim(0, 365)
    plt.ylim(1, 365)

    plt.plot(v)
    # plt.plot(w)
```

Sample plot for R
```R
library(ggplot2)
data("midwest")
ggplot(midwest, aes(x=area, y=poptotal)) + geom_point()
```

timtmok committed:
Support multiple formats for plot saving

Add SVG support

timtmok committed:
PDF support

timtmok committed:
JPEG support

timtmok committed:
simplify format check

timtmok committed:
cleanup

timtmok committed:
Formatting

timtmok committed:
Fix tests

seeM reviewed (commented):
Python changes LGTM!

timtmok committed:
Refactor format saving

Add format selector in dialog
Browse now sets directory save location
Add field for file name
Validate path at save time
Show error if saving failed

seeM reviewed (approved):
LGTM

timtmok committed:
Validate plot name
